pkg/util/util.go

```diff
@@ -22,3 +30,64 @@ func StringMatchRegexSlice(s string, re []string) bool {
 	}
 	return false
 }
+
+// ReadPassword reads a password from the terminal and restores the terminal state on interruption
+func ReadPassword(fd int) ([]byte, error) {
+	// Store and restore the terminal status on interruptions to
+	// avoid that the terminal remains in the password state
+	// This is necessary as for https://github.com/golang/go/issues/31180
+
+	/* WARNING: This commit contains two suggestions and should as such not be commited !! */
+	reemitSolution := true
+
+	/* Suggestion 1: Re-emit the interrupt signal, but restore the terminal in-between*/
+	if reemitSolution {
+		oldState, _ := term.GetState(fd)
+		interruptChannel := make(chan os.Signal, 1)
+		signal.Notify(interruptChannel, os.Interrupt)
+		defer signal.Stop(interruptChannel)
+		defer close(interruptChannel)
+
+		go func() {
+			for range interruptChannel {
+				if oldState != nil {
+					term.Restore(fd, oldState)
+				}
+				signal.Stop(interruptChannel)
+				syscall.Kill(syscall.Getpid(), syscall.SIGINT)
```

Comment:
Suggestion 1:

This works in a top-level application; it’s fairly suspect in library code. E.g. if there is anything else registered via `signal.Notify`, it will now see the signal twice.

Also, even with this approach, I think we _need_ to interrupt `term.ReadPassword`, because there could be some other party with a signal handler; in that case even this `syscall.Kill` might never terminate the `term.ReadPassword`.

The good thing about this approach is that the exit code of the process actually reflects the signal. (BTW that doesn’t actually happen directly here, there is _still_ a Go signal handler installed, but it does ~the same thing, uninstalling the handler and re-sending the signal.)

Reply:
My main concern is that there is no way of determining if there are other signal handlers for SIGINT present that attempt to gracefully terminate the application. This signal handler re-emits SIGINT and kills them. I could not find anything in `os/signal` that allows us to check, if there are other handlers present.

This makes this solution undesirable, as it would kill possible other signal handlers, apart from your points.